client/core/core.go

```diff
@@ -3582,15 +3582,48 @@ func (c *Core) resumeTrades(dc *dexConnection, trackers []*trackedTrade) assetMa
 					notifyErr(SubjectMatchStatusError, "Match %s for order %s is in state %s, but has no maker swap contract.", dbMatch.Side, tracker.token(), dbMatch.Status)
 					continue
 				}
+				// Obtaining AuditInfo will fail if it's unmined AND gone from
+				// mempool, or the wallet is otherwise not ready. Note that this
+				// does not actually audit the contract's value, recipient,
+				// expiration, or secret hash (if maker), as that was already
+				// done when it was initially stored as CounterScript.
 				auditInfo, err := wallets.toWallet.AuditContract(counterSwap, counterContract)
 				if err != nil {
-					c.log.Debugf("AuditContract error for match %v status %v, refunded = %v, revoked = %v: %v",
-						match.id, match.MetaData.Status, len(match.MetaData.Proof.RefundCoin) > 0, match.MetaData.Proof.IsRevoked(), err)
-					match.swapErr = fmt.Errorf("audit error, order %s, match %s: %w", tracker.ID(), match.id, err)
-					match.MetaData.Proof.SelfRevoked = true
-					notifyErr(SubjectMatchRecoveryError, "Error auditing counter-party's swap contract (%v) during swap recovery on order %s: %v",
-						tracker.token(), coinIDString(wallets.toAsset.ID, counterSwap), err)
-					continue
+					contractStr := coinIDString(wallets.toAsset.ID, counterSwap)
+					c.log.Warnf("Starting search for counterparty contract %v (%s)", contractStr, unbip(wallets.toAsset.ID))
+					// Start the audit retry waiter. Set swapErr to block tick
+					// actions like counterSwap.Confirmations checks while it is
+					// searching since matchTracker.counterSwap is not yet set.
+					match.swapErr = fmt.Errorf("audit in progress, please wait") // don't frighten the users
+					go func(tracker *trackedTrade, match *matchTracker) {
+						auditInfo, err := tracker.searchAuditInfo(match, counterSwap, counterContract)
+						tracker.mtx.Lock()
+						defer tracker.mtx.Unlock()
+						if err != nil {
+							match.swapErr = fmt.Errorf("audit error: %w", err)
+							c.log.Debugf("AuditContract error for match %v status %v, refunded = %v, revoked = %v: %v",
+								match.id, match.MetaData.Status, len(match.MetaData.Proof.RefundCoin) > 0,
+								match.MetaData.Proof.IsRevoked(), err)
+							notifyErr(SubjectMatchRecoveryError, "Error auditing counter-party's swap contract (%s %v) during swap recovery on order %s: %v",
+								unbip(wallets.toAsset.ID), contractStr, tracker.token(), err)
+							// If not already revoked, set it as self-revoked.
+							if match.MetaData.Proof.IsRevoked() {
+								return
+							}
+							match.MetaData.Proof.SelfRevoked = true
+							err = tracker.db.UpdateMatch(&match.MetaMatch)
+							if err != nil {
+								tracker.dc.log.Errorf("Error updating database for match %v: %v", match.id, err)
+							}
+							return
```

Comment:
I'm considering if it is best NOT to self-revoke it at all.  With swapErr still set, there won't be any attempted swaps or redeems, but the user could restart dexc if they fixed something with their rig and it would only retry if we did not revoke.  The server will eventually send a revoke_match or the client will discover it revoked later on 'connect'...